scripts/featured

```diff
@@ -0,0 +1,562 @@
+        self.subscriber_map = dict() # subscriber <-> fd map
+        self.advanced_boot = False
+        if swsscommon.RestartWaiter.isAdvancedBootInProgress(self.state_db_conn):
+            self.advanced_boot = True
+            swsscommon.RestartWaiter.waitAdvancedBootDone()
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect(wait_for_init=True, retry_on=True)
+        self.selector = swsscommon.Select()
+        syslog.syslog(syslog.LOG_INFO, 'ConfigDB connect success')
+
+        # Load DEVICE metadata configurations
+        self.device_config = {}
+        self.device_config['DEVICE_METADATA'] = self.config_db.get_table('DEVICE_METADATA')
+
+        # Load feature state table
+        feature_state_table = Table(self.state_db_conn, FEATURE_TBL)
+
+        # Intialize Feature Handler
+        self.feature_handler = FeatureHandler(self.config_db, feature_state_table, self.device_config, self.advanced_boot)
+        self.feature_handler.handle_adv_boot()
+
+    def subscribe(self, dbconn, table, callback, pri):
+        try:
+            if table not in self.callbacks:
+                self.callbacks[table] = []
+                subscriber = swsscommon.SubscriberStateTable(dbconn, table, swsscommon.TableConsumable.DEFAULT_POP_BATCH_SIZE, pri)
+                self.selector.addSelectable(subscriber) # Add to the Selector
+                self.subscriber_map[subscriber.getFd()] = (subscriber, table) # Maintain a mapping b/w subscriber & fd
+
+            self.callbacks[table].append(callback)
+        except Exception as err:
+            syslog.syslog(syslog.LOG_ERR, "Subscribe to table {} failed with error {}".format(table, err))
+
+    def register_callbacks(self):
+        def make_callback(func):
+            def callback(table, key, op, data):
+                return func(key, op, data)
+            return callback
+
+        self.subscribe(self.cfg_db_conn, FEATURE_TBL,
+                       make_callback(self.feature_handler.handler), HOSTCFGD_MAX_PRI)
+
+        self.subscribe(self.appl_db_conn, PORT_TBL,
+                       make_callback(self.feature_handler.port_listener), HOSTCFGD_MAX_PRI-1)
+
+    def render_all_feature_states(self):
+        features = self.config_db.get_table(FEATURE_TBL)
+        self.feature_handler.sync_state_field(features)
+
+    def start(self, init_time):
+        while True:
+            state, selectable_ = self.selector.select(DEFAULT_SELECT_TIMEOUT)
+
+            if state == self.selector.TIMEOUT:
+                if int(time.time() - init_time) > PORT_INIT_TIMEOUT_SEC:
+                    # if the delayed services are not enabled until PORT_INIT_TIMEOUT_SEC, enable them
+                    self.feature_handler.handle_port_table_timeout()
+                continue
+            elif state == self.selector.ERROR:
+                syslog.syslog(syslog.LOG_ERR, "error returned by select")
+                continue
+
+            fd = selectable_.getFd()
```

Comment:
>getFd [](http://example.com/codeflow?start=29&length=5)

The old implementation does not use `getFd()`. Could you explain why you need it? Is it a new requirement?

Reply:
The old implementation was using ConfigDbConnector. Featured is using SubscriberStateTable. 

I'm saving a map b/w SubscriberStateTable <-> fd and so that when When select returns, i can use the fd to find the corresponding Table and act accordingly.